arm-visualstudio/2014-04-01-preview/swagger/Csm.json

```diff
@@ -0,0 +1,867 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Visual Studio Resource Provider Client",
+    "version": "2014-04-01-preview",
+    "description": "Use these APIs to manage Visual Studio Team Services resources through the Azure Resource Manager. All task operations conform to the HTTP/1.1 protocol specification and each operation returns an x-ms-request-id header that can be used to obtain information about the request. You must make sure that requests made to these resources are secure. For more information, see https://docs.microsoft.com/en-us/rest/api/index."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/providers/Microsoft.VisualStudio/operations": {
+      "get": {
+        "tags": [
+          "accountResourceOperations"
+        ],
+        "summary": "GetOperationResources",
+        "description": "",
+        "operationId": "getOperationResources_List",
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation succeeded. The response contains the possible operations on the Microsoft.VisualStudio resource provider.",
+            "schema": {
+              "$ref": "#/definitions/ResourceOperationsDescriptorWrapper"
+            }
+         }
+        },
+        "x-ms-examples": {
+            "getOperationResources_List": {
+                "$ref": "../examples/GetOperations.json"
+            }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}": {
+      "get": {
+        "tags": [
+          "accountResourceGroup"
+        ],
+        "summary": "GetResources",
+        "description": "Gets the Visual Studio Team Services account resources linked to the specified Azure subscription.",
+        "operationId": "getResources_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "resourceGroupName",
+            "description": "Name of the resource group.",
+            "required": true,
+            "type": "string"
+          },
```

Comment:
**Moving resourceGroupName to global section**

Given that many operation paths uses `resourceGroupName`, it is recommended to move this definition to global parameters section (i.e. the section where subscriptionId and apiVersion is defined) 

```
    "resourceGroupNameParameter": {
      "name": "resourceGroupName",
      "in": "path",
      "required": true,
      "type": "string",
      "x-ms-parameter-location": "method",
      "description": "Name of the Resource group within the Azure subscription."
    },
```

and refer it from the parameter section of applicable operations as:

          {
            "$ref": "#/parameters/ResourceGroupNameParameter"
          },